### PR TITLE
feat: use new SDK standard tests framework

### PR DESCRIPTION
**Pull request by kgpayne**

### CODE REVIEW

**Comment by kgpayne** [comment]
@MeltanoLabs/engineering any ideas why `mypy` might be failing here 🤔 The `Property.default` type hint is `default: _JsonValue | None = None` so not sure why its not happy:

```python
_JsonValue: TypeAlias = Union[
    str,
    int,
    float,
    bool,
    list,
    dict,
    None,
]
```

Maybe that last `None` should be a `Type[None]` 🤷‍♂️

**Comment by edgarrmondragon** [comment]
> @MeltanoLabs/engineering any ideas why `mypy` might be failing here 🤔 The `Property.default` type hint is `default: _JsonValue | None = None` so not sure why its not happy

@kgpayne Try updating mypy

File: target_csv/tests/test_core.py [hunk 1]
```diff
@@ -1,25 +0,0 @@
...
-}
-
-
-# Run standard built-in target tests from the SDK:
-def test_standard_target_tests():
-    """Run standard target tests from the SDK."""
-    tests = get_standard_target_tests(
-        TargetCSV,
-        config=SAMPLE_CONFIG,
-    )
```
Comment (aaronsteers) [inline target_csv/tests/test_core.py:20]: @kgpayne - can you confirm if this is backwards compatible? What would happen if users bump the SDK without changing this test strategy? Ideally we would not break existing implementations - either we run all new tests, or run some subset - but we definitely would not want all taps and targets to have their tests broken with their next SDK version bump.

Comment (aaronsteers) [reply target_csv/tests/test_core.py]: I decided this was easier to just test by reverting the dropped tests:

- https://github.com/MeltanoLabs/target-csv/pull/5/commits/b1dd55455f0ce804cdc08c44a88f7c13eb42e769

Comment (aaronsteers) [reply target_csv/tests/test_core.py]: ✅ Legacy tests still pass! 🎉

**Comment by edgarrmondragon** [comment]
Superseded by changes in #49

**Pull request closed without merging**